primus_vk.cpp

```diff
@@ -83,6 +84,28 @@ std::shared_ptr<void> libvulkan(dlopen("libvulkan.so.1", RTLD_NOW), dlclose);
 // #define VK_CHECK_RESULT(x) if(x != VK_SUCCESS){printf("Error %d, in %d\n", x, __LINE__);}
 
 
+void GetEnvVendorDeviceIDs(std::string env, uint32_t &vendor, uint32_t &device)
+{
+  char *envstr = getenv(env.c_str());
+  if(envstr != nullptr){
+    std::stringstream ss(envstr);
+    std::string item;
+    std::vector<std::string> elems;
+    std::vector<uint32_t> hexnums(2);
+    int i = 0;
+    while(std::getline(ss, item, ':')) {
+      uint32_t num = 0;
+      std::stringstream _ss;
+      _ss << std::hex << item;
+      _ss >> num;
+      hexnums[i] = num;
+      ++i;
```

Comment:
cant this loop do an out-of-bounds access on the vector if the string contains more than one colon?

Reply:
Yep, fixed!